staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -18,10 +18,81 @@ package runtime
 
 import (
 	"bytes"
-	"encoding/json"
+	gojson "encoding/json"
 	"errors"
+	"fmt"
+	"reflect"
+
+	cbordirect "k8s.io/apimachinery/pkg/runtime/serializer/cbor/direct"
+	"k8s.io/apimachinery/pkg/util/json"
+	"sigs.k8s.io/structured-merge-diff/v4/value"
+
+	cbor "github.com/fxamacker/cbor/v2"
 )
 
+// RawExtension intentionally avoids implementing value.UnstructuredConverter for now because the
+// signature of ToUnstructured does not allow returning an error value in cases where the conversion
+// is not possible (content type is unrecognized or bytes don't match content type).
+func (re RawExtension) toUnstructured() (interface{}, error) {
+	if re.Raw != nil {
+		contentType, err := re.rawContentType()
+		if err != nil {
+			return nil, err
+		}
+
+		switch contentType {
+		default:
+			return nil, fmt.Errorf("cannot convert RawExtension with unrecognized content type %q to unstructured", contentType)
```

Comment:
Detect type here?

Reply:
This is switching on the inferred type already. If the type can't be detected, `rawContentType` returns a non-nil error before getting here. I originally wrote this case in the context where we have an optional explicit ContentType field, so without that it's not possible to reach this error.